beekeeper-path-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/filter/EventTypeTableListenerEventFilter.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (C) 2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.scheduler.apiary.filter;
+
+import com.expedia.apiary.extensions.receiver.common.event.EventType;
+import com.expedia.apiary.extensions.receiver.common.event.ListenerEvent;
+import org.springframework.stereotype.Component;
+import static com.expediagroup.beekeeper.core.model.LifeCycleEventType.UNREFERENCED;
+import static com.expediagroup.beekeeper.core.model.LifeCycleEventType.EXPIRED;
+
+import java.util.Map;
+
+@Component
+public class EventTypeTableListenerEventFilter implements ListenerEventFilter {
+
+  @Override
+  public boolean filter(ListenerEvent listenerEvent) {
+    if (listenerEvent == null) {
+      return true;
+    }
+
+    Map<String, String> tableParameters = listenerEvent.getTableParameters();
+    if (tableParameters == null) {
+      return true;
+    }
+
+    Boolean isUnreferenced = UNREFERENCED.getBoolean(tableParameters);
+    Boolean isExpired = EXPIRED.getBoolean(tableParameters);
+
+    EventType eventType = listenerEvent.getEventType();
```

Comment:
Would be good to agree on what these scenarios are, in my mind `CREATE_TABLE` and `ALTER_TABLE` would insert a new expired row, and `ADD_PARTITION`, `ALTER_PARTITION` and `ALTER_TABLE` would update expired rows with a new deletion timestamp.

I think that `CREATE_TABLE` wouldn't necessarily have the required parameter we need on the table to check that we need to monitor it. We could either (1) require this parameter is added to the database and make sure Apiary emits these parameters or (2) maintain a list databases on Beekeeper which are marked to be monitored for expired data. (1) is definitely the nicer option but will need some extra work outside of Beekeeper.

Reply:
Good question. It depends a bit how we expect this all to be used. Are we going to say there will be a "scratch" database and *all* tables in this DB will have automatic expiry? I'm not sure which DB events we would listen to - there are create and drop events for DB's but I think that's about it. So we might have to go with option 2.

Reply:
It was my understanding that Hive can emit database parameters along with table events. If this is true, and we can get the Apiary listener sending this information along with create table events, then we could check for this database parameter without the need for maintaining a list on the service itself, or for monitoring create/drop database events.

Reply:
Ah right, yes, we'd need to try it out but that could work.